src/clr-angular/data/tree-view/tree-node.ts

```diff
@@ -80,7 +80,20 @@ export class ClrTreeNode<T> implements OnInit, OnDestroy {
       // I'm completely stuck, we have to hack into private properties until either
       // https://github.com/angular/angular/issues/14935 or https://github.com/angular/angular/issues/15998
       // are fixed
-      this._model = (<any>injector).view.context.clrModel;
+      // This is for non-ivy implementations
+      if ((<any>injector).view) {
+        this._model = (<any>injector).view.context.clrModel;
+      } else {
+        // Ivy has a potentially unpredictable internal API to get a parent's model, so a new hack!
+        // tslint:disable-next-line
+        for (const i in (<any>injector)._lView) {
```

Comment:
hmm... So I really would like to avoid this kind of fix as it's first accessing the internal properties by going through all these internal properties. I would like to investigate other options that could help circumvent this.

Reply:
I would too, but for the immediate moment I don't have time to investigate other options that aren't breaking changes.

Reply:
I was actually looking into this issue for a while now. If we find any solution, I think it could exist in our internal component `RecursiveChildren`.

Reply:
As we talked, we could just resort to 
```
export const LVIEW_CONTEXT_INDEX = 8;
....
this._model = (<any>injector)._lView[LVIEW_CONTEXT_INDEX].clrModel;
```